src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java

```diff
@@ -289,4 +290,15 @@ private boolean matchesChipsField(String field, String chipsField) {
         return field.trim().equalsIgnoreCase(chipsField.trim());
     }
 
-}
\ No newline at end of file
+    public void validateAddressesMultipleFlagsUpdate(HttpServletRequest request, List<ApiError> errorList, OfficerFilingDto dto, AppointmentFullRecordAPI appointmentFullRecordAPI) {
+        if (Boolean.TRUE.equals(dto.getIsHomeAddressSameAsServiceAddress()) && Boolean.TRUE.equals(dto.getIsServiceAddressSameAsRegisteredOfficeAddress())) {
+            createValidationError(request, errorList, apiEnumerations.getValidation(ValidationEnum.ADDRESS_LINKS_MULTIPLE_FLAGS));
+        }
```

Comment:
I think we may need to add && dto.getIsOtherAddressSameAsServiceAddress to line 297 and 300 (changing Other to RegisteredOffice and Home respectively. So only erroring if we are setting one of the flags and the other stays the same. That is probably not in the ACs so let's discuss in dev standup.